apps/skilavottord/ws/src/app/modules/accessControl/accessControl.service.ts

```diff
@@ -1,69 +1,132 @@
+import { AccessControlModel } from './accessControl.model'
+import { LOGGER_PROVIDER } from '@island.is/logging'
 import { Inject, Injectable } from '@nestjs/common'
-
 import type { Logger } from '@island.is/logging'
-import { LOGGER_PROVIDER } from '@island.is/logging'
-
-import { Role } from '../auth'
-import { AccessControlModel } from './accessControl.model'
+import { InjectModel } from '@nestjs/sequelize'
 import {
   CreateAccessControlInput,
   DeleteAccessControlInput,
   UpdateAccessControlInput,
 } from './accessControl.input'
+import { RecyclingPartnerModel } from '../recyclingPartner/recyclingPartner.model'
 
 @Injectable()
 export class AccessControlService {
-  constructor(@Inject(LOGGER_PROVIDER) private logger: Logger) {}
+  constructor(
+    @InjectModel(AccessControlModel)
+    private accessControlModel: typeof AccessControlModel,
+    @Inject(LOGGER_PROVIDER) private logger: Logger,
+  ) {}
 
   async findAll(): Promise<AccessControlModel[]> {
     this.logger.info('---- Starting findAll Access Controls ----')
+    try {
+      const res = await AccessControlModel.findAll({
+        include: [
+          {
+            model: RecyclingPartnerModel,
+          },
+        ],
+      })
+      this.logger.info(
+        'findAll-recyclingUsers result:' + JSON.stringify(res, null, 2),
+      )
+      return res
+    } catch (error) {
+      this.logger.error('error finding all recyclingUsers:' + error)
+    }
+  }
 
-    // TODO replace mock data with actual db query
-    const res = await Promise.resolve([
-      {
-        nationalId: '1234567777',
-        name: 'Gervimaður',
-        role: Role.recyclingCompany,
-        partnerId: '8888888888',
-      },
-      {
-        nationalId: '1234567888',
-        name: 'Gervimaður2',
-        role: Role.recyclingCompany,
-        partnerId: '9999999999',
-      },
-      {
-        nationalId: '1234567899',
-        name: 'Gervimaður3',
-        role: Role.recyclingCompany,
-        partnerId: '9999999999',
-      },
-    ] as AccessControlModel[])
-    return res
+  async findOne(nationalId: string): Promise<AccessControlModel> {
+    this.logger.info('find one access user...')
+    try {
+      const res = await this.accessControlModel.findOne({
+        where: { nationalId: nationalId },
+        include: [
+          {
+            model: RecyclingPartnerModel,
+          },
+        ],
+      })
+      this.logger.info(
+        'findOne-accessControl result:' + JSON.stringify(res, null, 2),
+      )
+      // null if not found
+      return res
+    } catch (error) {
```

Comment:
Why do we need those try catches?

This goes for every single database querying in here... if there's an error, it should be automatically logged by the logging provider ... and it wouldn't swallow the error like this is doing, so the client would actually see that something is wrong by automatically transforming the error to a graphql error which the error link middleware takes care of rendering.

The code could actually just return the promise.
```typescript
return this.accessControlModel.findOne({...})
```